File: python/sparklabs/batch/sales/rdd/top_selling_products.py
from sparklabs.batch.sales.rdd import utils


def run() -> None:
    rows = (
        utils.order_items.map(lambda item: (item["product_id"], 1))
        .reduceByKey(lambda left, right: left + right)
        .sortBy(lambda item: item[1], ascending=False)
        .collect()
    )
    for row in rows:
        print(row)


if __name__ == "__main__":
    run()

File: scala/src/main/scala/io/devorbit/sparklabs/batch/sales/rdd/TopSellingProducts.scala
package io.devorbit.sparklabs.batch.sales.rdd

object TopSellingProducts extends App {
  RddUtils.orderItems
    .map(item => (item.product_id, 1))
    .reduceByKey(_ + _)
    .sortBy(_._2, ascending = false)
    .collect()
    .foreach(println)
}
